### PR TITLE
Fix: Fix for physical when budget is reached

### PR DESCRIPTION
**Pull request by AveryQi115**
Continue on executing physical rules when budget limit is reached.

### CODE REVIEW

**Review by skyzh — APPROVED** [review]
🤪 thanks for the fix!

File: optd-sqlplannertest/tests/tpch.planner.sql [hunk 1]
```diff
@@ -364,72 +364,69 @@ PhysicalLimit { skip: 0, fetch: 100 }
...
-                                                                                                        │   └── PhysicalScan { table: nation }
-                                                                                                        └── PhysicalProjection { exprs: [ #0 ] }
-                                                                                                            └── PhysicalFilter
-                                                                                                                ├── cond:Eq
-                                                                                                                │   ├── #1
-                                                                                                                │   └── "AFRICA"
-                                                                                                                └── PhysicalProjection { exprs: [ #0, #1 ] }
-                                                                                                                    └── PhysicalScan { table: region }
+        └── PhysicalNestedLoopJoin
```
Comment (skyzh) [inline optd-sqlplannertest/tests/tpch.planner.sql:367]: NLJ not optimized into hash join?

Comment (skyzh) [reply optd-sqlplannertest/tests/tpch.planner.sql:367]: or something wrong with the cost model that it assumes NLJ costs less than hash join

Comment (AveryQi115) [reply optd-sqlplannertest/tests/tpch.planner.sql:367]: Oh it's due to our physical conversion rule of hashjoin. It only takes one pair of equal keys so I assume nlj is more efficient than multiple layers of projection and hashjoin. See the previous solution, it contains a lot more projection and hash join. I'll merge this pr first and modify current hash join later.

**Pull request merged**